test_scripts/API/VehicleData/StabilityControlsStatus/017_SubscribeVehicleData_StabilityControlsStatus_and_other_parameters_invalid_hmi_response.lua

```diff
@@ -0,0 +1,104 @@
+--[[ Local variables ]]
+local vdNameInvalidRes = "gps"
+local vdNameSuccess = "stabilityControlsStatus"
+local mobileRequest = {
+  [vdNameSuccess] = true,
+  [vdNameInvalidRes] = true
+}
+local hmiRequest = mobileRequest
+
+local hmiResponseEmpty = {
+  [vdNameInvalidRes] = {},
+  [vdNameSuccess] = common.getSubscribeVehicleDataHmiResponse("SUCCESS", vdNameSuccess)
+}
+local hmiResponseInvalidValues = {
+  [vdNameInvalidRes] = { resultCode = "SUCCESS", dataType = "invalidType" },
+  [vdNameSuccess] = common.getSubscribeVehicleDataHmiResponse("SUCCESS", vdNameSuccess)
+}
+local hmiResponseAbsent = {
+  [vdNameSuccess] = common.getSubscribeVehicleDataHmiResponse("SUCCESS", vdNameSuccess)
+}
+local hmiResponseIncorrectType = {
+  -- mandatory parameter 'dataType' is missing
+  [vdNameInvalidRes] = { resultCode = "SUCCESS" },
+  [vdNameSuccess] = common.getSubscribeVehicleDataHmiResponse("SUCCESS", vdNameSuccess)
+}
+local hmiResponseNotAStructure = {
+  [vdNameInvalidRes] = 1,
+  [vdNameSuccess] = common.getSubscribeVehicleDataHmiResponse("SUCCESS", vdNameSuccess)
+}
+local mobileResponse = { success = false, resultCode = "GENERIC_ERROR" }
+
+--[[ Scenario ]]
+common.Title("Preconditions")
+common.Step("Clean environment", common.preconditions)
+common.Step("Start SDL, init HMI, connect default mobile", common.start)
+common.Step("Register App", common.registerApp)
+common.Step("PTU", common.policyTableUpdate, { common.ptUpdate })
+common.Step("Activate App", common.activateApp)
+
+common.Title("Test")
+common.Step("Subscribe on VehicleData, invalid HMI response: empty structure",
+  common.processSubscribeVD, { mobileRequest, hmiRequest, hmiResponseEmpty, mobileResponse })
+common.Step("Ignore OnVehicleData with " .. vdNameInvalidRes .. " data",
+  common.checkNotificationIgnored, { vdNameInvalidRes })
+common.Step("Ignore OnVehicleData with " .. vdNameSuccess .. " data",
+  common.checkNotificationIgnored, { vdNameSuccess })
+
+common.Step("Subscribe on VehicleData, invalid HMI response: invalid values",
+  common.processSubscribeVD, { mobileRequest, hmiRequest, hmiResponseInvalidValues, mobileResponse })
+common.Step("Ignore OnVehicleData with " .. vdNameInvalidRes .. " data",
+  common.checkNotificationIgnored, { vdNameInvalidRes })
+common.Step("Ignore OnVehicleData with " .. vdNameSuccess .. " data",
+  common.checkNotificationIgnored, { vdNameSuccess })
+
+common.Step("Subscribe on VehicleData, invalid HMI response: structure is absent",
+  common.processSubscribeVD, { mobileRequest, hmiRequest, hmiResponseAbsent, mobileResponse })
+common.Step("Ignore OnVehicleData with " .. vdNameInvalidRes .. " data",
+  common.checkNotificationIgnored, { vdNameInvalidRes })
+common.Step("Expect OnVehicleData with " .. vdNameSuccess .. " data",
+  common.checkNotificationSuccess, { vdNameSuccess })
+common.Step("Unsubscribe from " .. vdNameSuccess .. " VehicleData",
+  common.processRPCSubscriptionSuccess, { "UnsubscribeVehicleData", vdNameSuccess })
+
```

Comment:
@mvorobio I would prop[ose to move steps related to this case at the end of the list of the script checks and remove last step as redundant with goal to avoid creation of dependency on UnsubscribeVehicleData RPC

Reply:
@aderiabin please check https://github.com/LuxoftSDL/sdl_atf_test_scripts/pull/15/commits/fce4eef1e5deaf54aa76b24f447c76c0a48bc417